modules/host/yubico.nix

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  cfg = config.ghaf.host.passwordless_authentication;
+in {
+  options.ghaf.host.passwordless_authentication = {
+    enable = lib.mkEnableOption "Host Yubikeys Support";
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [
+      # For generating and debugging Yubikeys
+      pkgs.pam_u2f
+    ];
+
+    security.pam.services = {
+      login.u2fAuth = true;
+      sudo.u2fAuth = true;
+    };
+    security.pam.u2f.authFile = "/etc/u2f_mappings";
```

Comment:
Please add a combination of `assert`, `assertMsg` and `builtins.pathExists <path>` so that the user gets a proper error message if the `security.pam.u2f.authFile` has not been created by the user.

https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20assertmsg&type=code